pkg/concurrency/runner.go

```diff
@@ -0,0 +1,89 @@
+package concurrency
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"sync"
+)
+
+// Runner is a function that runs a task.
+type Runner func(context.Context) error
+
+// RunnerManager is a manager for runners. It runs all runners in parallel and
+// waits for all runners to finish. If any runner returns, the RunnerManager
+// will stop all other runners and return any error.
+type RunnerManager struct {
```

Comment:
nit, I'd rather prefer the names used by js and .NET

async.All(...taskList)/async.Any(...taskList)

Reply:
Not too sure what you mean- do you mean renaming `RunnerManager` to `async`?

Reply:
basically, and use an approach more functional-like instead of object-like

async.Any(runners[]...) does all the thing you need to